pkg/pillar/cmd/baseosmgr/forcefallback.go

```diff
@@ -40,6 +40,19 @@ func handleZedAgentStatusImpl(ctxArg interface{}, key string,
 	ctxPtr := ctxArg.(*baseOsMgrContext)
 	status := statusArg.(types.ZedAgentStatus)
 	handleForceFallback(ctxPtr, status)
+
+	if ctxPtr.currentUpdateRetry != status.BaseosUpdateCounter {
```

Comment:
Move logic to handler

Reply:
If we are under upgrade/testing, just update configUpdateRetry
If we are in failing state (create function for checking), save currentUpdateRetry, update configUpdateRetry, issue upgrade
Else update all

Reply:
Check here for configUpdateRetry to reduce checks

Reply:
How to avoid re-retry? Seems, we must save somewhere in case of errors

Reply:
Without saving of configUpdateRetry, we will have a loop of re-upgrades. We will do the same check after reboot-failed-reboot.

Reply:
Probably, we can save new value in case of updating/testing without updating of currentUpdateRetry?